src/config.rs

```diff
@@ -0,0 +1,399 @@
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default)]
+#[serde(rename_all = "PascalCase")]
+pub struct Config {
+    /// The username or UID which is a platform-specific structure
+    /// that allows specific control over which user the process run as. This acts as a default value to use when the value is
+    /// not specified when creating a container. For Linux based
+    /// systems, all of the following are valid: `user`, `uid`,
+    /// `user:group`, `uid:gid`, `uid:group`, `user:gid`. If `group`/`gid` is
+    /// not specified, the default group and supplementary groups
+    /// of the given `user`/`uid` in `/etc/passwd` from the container are
+    /// applied.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub user: Option<String>,
+
+    /// A set of ports to expose from a container running this
+    /// image. Its keys can be in the format of: `port/tcp`, `port/udp`,
+    /// `port` with the default protocol being `tcp` if not specified.
+    /// These values act as defaults and are merged with any
+    /// specified when creating a container.
+    #[serde(skip_serializing_if = "HashMap::is_empty")]
+    pub exposed_ports: HashMap<String, serde_json::Value>,
+
+    /// Entries are in the format of `VARNAME=VARVALUE`.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub env: Vec<String>,
+
+    /// Default arguments to the entrypoint of the container.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub cmd: Vec<String>,
+
+    /// A list of arguments to use as the command to execute when
+    /// the container starts..
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub entrypoint: Vec<String>,
+
+    /// A set of directories describing where the process is likely write data specific to a container instance.
+    #[serde(skip_serializing_if = "HashMap::is_empty")]
+    pub volumes: HashMap<String, serde_json::Value>,
+
+    /// Sets the current working directory of the entrypoint
+    /// process in the container.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub working_dir: Option<String>,
+
+    /// The field contains arbitrary metadata for the container.
+    /// This property MUST use the [annotation rules](https://github.com/opencontainers/image-spec/blob/v1.0/annotations.md#rules).
+    #[serde(skip_serializing_if = "HashMap::is_empty")]
+    pub labels: HashMap<String, String>,
+
+    /// The field contains the system call signal that will be sent
+    /// to the container to exit. The signal can be a signal name
+    /// in the format `SIGNAME`, for instance `SIGKILL` or `SIGRTMIN+3`.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stop_signal: Option<String>,
+}
+
+/// The rootfs key references the layer content addresses used by the image.
+/// This makes the image config hash depend on the filesystem hash.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default)]
+pub struct Rootfs {
+    /// MUST be set to `layers`.
```

Comment:
We should probably have a `const` defined for `layers`

Reply:
Nice suggestion, fixed it